app/services/edge_cache/bust/fastly.rb

```diff
@@ -1,24 +1,37 @@
 module EdgeCache
   class Bust
     class Fastly
+      # [@forem/systems] Fastly-enabled Forems don't need "flexible" domains.
       def self.call(path)
-        # TODO: (Alex Smith) - It would be "nice to have" the ability to use the
-        # Fastly gem here instead of custom API calls.
+        api_key = ApplicationConfig["FASTLY_API_KEY"]
+        return fastly_purge(api_key, path) if FeatureFlag.enabled?(:fastly_http_purge)
 
-        # @forem/systems Fastly-enabled forems don't need "flexible" domains.
-        HTTParty.post(
-          "https://api.fastly.com/purge/https://#{URL.domain}#{path}",
-          headers: {
-            "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"]
-          },
-        )
-        HTTParty.post(
-          "https://api.fastly.com/purge/https://#{URL.domain}#{path}?i=i",
-          headers: {
-            "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"]
-          },
-        )
+        fastly_post(api_key, path)
       end
+
+      def self.fastly_post(api_key, path)
+        urls(path).map do |url|
+          HTTParty.post("https://api.fastly.com/purge/#{url}", headers: { "Fastly-Key" => api_key })
+        end
+      end
+      private_class_method :fastly_post
+
+      def self.fastly_purge(api_key, path)
+        fastly = ::Fastly.new(api_key: api_key)
+
+        urls(path).map do |url|
+          fastly.purge(url)
```

Comment:
using [fastly-ruby gem here](https://github.com/fastly/fastly-ruby#efficient-purging). The [implementation is more or less what we would have written anyway](https://github.com/fastly/fastly-ruby/blob/acae3028a079f9bf0cbea4eaa349a0853441053d/lib/fastly/client.rb#L92-L107) and we get `Thread.current` [for free](https://github.com/fastly/fastly-ruby/blob/acae3028a079f9bf0cbea4eaa349a0853441053d/lib/fastly/client.rb#L111-L113)

Reply:
Great call! I'm so stoked 🔥

Reply:
@rhymes Is there a reason why we can't also use the gem in `fastly_post`? It seems odd to me to use `HTTParty` in one method and the gem in the other, but maybe I'm missing something.

Reply:
@citizen428 I thought about it but I decided to only change one thing at a time, since `HTTParty.post` works. Also the objective here is to remove that code, so I didn't bother checking if there were any differences :-)